tests/e2e/lib.sh

```diff
@@ -21,6 +21,55 @@ export QA_DEPLOY_WAIT_INFO="/tmp/wait-for-kubectl-object"
 # `envsubst`` before passing it to `env`.
 envsubst=$(command -v envsubst)
 
+# Detect stat variant (GNU vs BSD) and save the original stdin's device and inode
+# This is used by the kubectl wrapper function to detect stdin redirection
+_KUBECTL_WRAPPER_USE_GNU_STAT="true"
+_stdin_id=""
+if stat --version 2>&1 | grep "GNU coreutils" >/dev/null; then
```

Comment:
```suggestion
if stat --version 2>&1 | grep -q "GNU coreutils" >/dev/null; then
```

Not sure about the `coreutils` part... there are now efforts to replace various utilities with rust-written versions, so it would not surprise me if distros start shipping those instead of coreutils in the next few years...

Reply:
So what about this part of the comment?

> Not sure about the `coreutils` part... there are now efforts to replace various utilities with rust-written versions, so it would not surprise me if distros start shipping those instead of coreutils in the next few years...